java/ql/src/experimental/Security/CWE/CWE-552/UnsafeServletRequestDispatch.java

```diff
@@ -0,0 +1,82 @@
+public class UnsafeServletRequestDispatch extends HttpServlet {
+	private static final String BASE_PATH = "/pages";
+
+	@Override
+	protected void doGet(HttpServletRequest request, HttpServletResponse response)
```

Comment:
This example is too long for a qhelp file, mainly by repeating lots of boilerplate. For this purpose it would be enough to use a style like


```
// To check there won't be unexpected path traversal, we can use Path.normalize:
(code here)
// ... or alternatively ensure URL encoding is removed and then check for any `..` sequences ...
(code here)
// and so on
```

Basically bear in mind this is supposed to be easy to read, not a full test case.

Reply:
I've updated the test cases to be much more concise and easier to read. Please help to check whether further improvement is required. Thanks.